pkg/apis/extensions/types.go

```diff
@@ -208,7 +208,7 @@ type DeploymentSpec struct {
 	Selector map[string]string `json:"selector,omitempty"`
 
 	// Template describes the pods that will be created.
-	Template *api.PodTemplateSpec `json:"template,omitempty"`
+	Template api.PodTemplateSpec `json:"template"`
```

Comment:
I discussed this with @bgrant0607 yesterday. The work was half done and no one owns it. It's likely not going to get done before 1.2. Until that happens, generated documentation is confusing and incorrect (says this field is optional when it is actually not) and has been for months. The change back can be made in conversions when we have a new API version.